### PR TITLE
sci-libs/scikits_learn: version bump 0.22.1

### PR DESCRIPTION
**Pull request by TheChymera**
And PYTHON_COMPAT update to include 3_7

Package-Manager: Portage-2.3.94, Repoman-2.3.21
Signed-off-by: Horea Christian <chr@chymera.eu>

### CODE REVIEW

File: sci-libs/scikits_learn/scikits_learn-0.22.1.ebuild [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+
+PYTHON_COMPAT=( python3_{6,7} )
+
+inherit distutils-r1 flag-o-matic
+
+MYPN="${PN/scikits_/scikit-}"
+MYP="${MYPN}-${PV}"
+
+DESCRIPTION="Python modules for machine learning and data mining"
+HOMEPAGE="http://scikit-learn.org"
```
Comment (charIes17) [inline sci-libs/scikits_learn/scikits_learn-0.22.1.ebuild]: https://scikit-learn.org/

**Comment by TheChymera** [comment]
@charIes17 thanks for the hint. can you pull this? :)

**Comment by TheChymera** [comment]
@mgorny can you pull this as well? It's part of the Python 3.7 update.

File: sci-libs/scikits_learn/scikits_learn-0.22.1.ebuild [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+	sci-libs/scikits[${PYTHON_USEDEP}]
+	sci-libs/scipy[${PYTHON_USEDEP}]
+	virtual/blas:=
+	virtual/cblas:=
+"
+
+DEPEND="
+	dev-python/cython[${PYTHON_USEDEP}]
+	dev-python/numpy[lapack,${PYTHON_USEDEP}]
+	dev-python/setuptools[${PYTHON_USEDEP}]
```
Comment (mgorny) [inline sci-libs/scikits_learn/scikits_learn-0.22.1.ebuild]: Redundant.

File: sci-libs/scikits_learn/scikits_learn-0.22.1.ebuild [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+	dev-python/matplotlib[${PYTHON_USEDEP}]
+	dev-python/nose[${PYTHON_USEDEP}]
+	dev-python/numpy[lapack,${PYTHON_USEDEP}]
+	sci-libs/scikits[${PYTHON_USEDEP}]
+	sci-libs/scipy[${PYTHON_USEDEP}]
+	virtual/blas:=
+	virtual/cblas:=
+"
+
+DEPEND="
```
Comment (mgorny) [inline sci-libs/scikits_learn/scikits_learn-0.22.1.ebuild]: BDEPEND.

File: sci-libs/scikits_learn/scikits_learn-0.22.1.ebuild [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+	sci-libs/scipy[${PYTHON_USEDEP}]
+	virtual/blas:=
+	virtual/cblas:=
+"
+
+S="${WORKDIR}/${MYP}"
+
+python_prepare_all() {
+	# bug #397605
+	[[ ${CHOST} == *-darwin* ]] \
```
Comment (mgorny) [inline sci-libs/scikits_learn/scikits_learn-0.22.1.ebuild]: Move `&&` and `||` to EOL, so you don't need backslashes.

File: sci-libs/scikits_learn/scikits_learn-0.22.1.ebuild [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+		&& append-ldflags -bundle "-undefined dynamic_lookup" \
+		|| append-ldflags -shared
+
+	# scikits-learn now uses the horrible numpy.distutils automagic
+	export SCIPY_FCONFIG="config_fc --noopt --noarch"
```
Comment (mgorny) [inline sci-libs/scikits_learn/scikits_learn-0.22.1.ebuild]: Why do you need to export it? Why not an array?

File: sci-libs/scikits_learn/scikits_learn-0.22.1.ebuild [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+	distutils-r1_python_prepare_all
+}
+
+python_compile() {
+	distutils-r1_python_compile ${SCIPY_FCONFIG}
+}
+
+python_test() {
+	distutils_install_for_testing ${SCIPY_FCONFIG}
+	esetup.py install \
```
Comment (mgorny) [inline sci-libs/scikits_learn/scikits_learn-0.22.1.ebuild]: Why do you need to install it twice?

File: sci-libs/scikits_learn/scikits_learn-0.22.1.ebuild [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+			  --root="${T}/test-${EPYTHON}" \
+			  --no-compile ${SCIPY_FCONFIG}
+	pushd "${T}/test-${EPYTHON}/$(python_get_sitedir)" || die > /dev/null
+	#pytest --skip-network sklearn || die
+	pytest -m "not network" || die
```
Comment (mgorny) [inline sci-libs/scikits_learn/scikits_learn-0.22.1.ebuild]: -vv

File: sci-libs/scikits_learn/scikits_learn-0.22.1.ebuild [hunk 8]
```diff
@@ -0,0 +1,103 @@
...
+}
+
+python_install() {
+	distutils-r1_python_install ${SCIPY_FCONFIG}
+}
+
+python_install_all() {
+	find "${S}" -name \*LICENSE.txt -delete
```
Comment (mgorny) [inline sci-libs/scikits_learn/scikits_learn-0.22.1.ebuild]: ||die

**Comment by TheChymera** [comment]
@mgorny ok now?

File: sci-libs/scikits_learn/scikits_learn-0.22.1.ebuild [hunk 9]
```diff
@@ -0,0 +1,82 @@
...
+
+python_compile() {
+	distutils-r1_python_compile ${SCIPY_FCONFIG}
+}
+
+python_test() {
+	esetup.py install \
+		--root="${T}/test-${EPYTHON}" \
+		--no-compile ${SCIPY_FCONFIG}
+	pushd "${T}/test-${EPYTHON}/$(python_get_sitedir)" || die > /dev/null
```
Comment (mgorny) [inline sci-libs/scikits_learn/scikits_learn-0.22.1.ebuild]: There's no `popd`.

Comment (TheChymera) [reply sci-libs/scikits_learn/scikits_learn-0.22.1.ebuild]: @mgorny solved

File: sci-libs/scikits_learn/scikits_learn-0.22.1.ebuild (hunk 6 above)
Comment (mgorny) [reply sci-libs/scikits_learn/scikits_learn-0.22.1.ebuild]: I'm still confused as to why you need to do that. After all, this shouldn't make any difference compared to the contents of `${BUILD_DIR}/lib`.

Comment (TheChymera) [reply sci-libs/scikits_learn/scikits_learn-0.22.1.ebuild]: Might have something to do with the parameters. In any case, `distutils_install_for_testing` fails with this [build.log](https://github.com/gentoo/gentoo/files/4392273/build.log).

Comment (mgorny) [reply sci-libs/scikits_learn/scikits_learn-0.22.1.ebuild]: What about without any install command at all, just using build_dir?

Comment (TheChymera) [reply sci-libs/scikits_learn/scikits_learn-0.22.1.ebuild]: What do you mean by “just using `build_dir`”?

Comment (mgorny) [reply sci-libs/scikits_learn/scikits_learn-0.22.1.ebuild]: `${BUILD_DIR}/lib` to pytest

Comment (TheChymera) [reply sci-libs/scikits_learn/scikits_learn-0.22.1.ebuild]: No, that won't work at all, as it doesn't even find most of the tests. The closest I could get it to actually working is by using:

```
    #esetup.py install \
    #   --root="${T}/test-${EPYTHON}"
    #pushd "${T}/test-${EPYTHON}/$(python_get_sitedir)" || die > /dev/null
    #pytest -m "not network" -vv || die
    #popd > /dev/null
    distutils_install_for_testing
    ${EPYTHON} -m pytest -v "${BUILD_DIR}/test/lib/sklearn" || die
```

This fails with the following [build.log](https://github.com/gentoo/gentoo/files/4398081/build.log).

If I comment out the new version and uncomment the commented lines, it works. I would suggest we go with what works.

Comment (TheChymera) [reply sci-libs/scikits_learn/scikits_learn-0.22.1.ebuild]: @mgorny so, is it ok if I just go with:

```
    esetup.py install \
       --root="${T}/test-${EPYTHON}"
    pushd "${T}/test-${EPYTHON}/$(python_get_sitedir)" || die > /dev/null
    pytest -m "not network" -vv || die
    popd > /dev/null
```

That's about as standard as I could get it while still having it work.

**Comment by TheChymera** [comment]
@mgorny superseded by [7ac2026](https://gitweb.gentoo.org/repo/gentoo.git/commit/?id=7ac2026d1c2deeaa4ecc8ba6080351ebcd833649), which funnily enough and unlike this PR, is [actually broken](https://bugs.gentoo.org/716016).

**Comment by heroxbd** [comment]
@TheChymera Thanks anyway for the PR.  Sorry I did not take a look in time.

**Comment by rth** [comment]
Would anyone know what needs to happen to rename this package to the correct name of "scikit-learn" without the "s"? cf https://bugs.gentoo.org/665482

**Pull request closed without merging**